docker/BuildJekyll.Dockerfile

```
@@ -0,0 +1,20 @@
+FROM ruby:3.0.0
+
+WORKDIR /srv/jekyll
+COPY ./static/Gemfile .
+
+ARG uid
+ARG uname
+ARG gid
+ARG gname
+
+RUN groupadd -g $gid $gname
+RUN useradd -ms /bin/bash -u $uid -g $gid $uname
```

The names aren't important (as you may recall, the file metadata only stores the ID), and there's only one setting of `gid` and `uid` that make any sense here, so I wouldn't take them as arguments:

```suggestion
RUN groupadd -g 1000 jekyll
RUN useradd -ms /bin/bash -u 1000 -g 1000 jekyll
```

---

Okay! It makes sense that the group and user name isn't actually stored, but don't we want to make sure that whatever user id is on the host machine also gets placed into the image? Why is 1000 the only uid/gid that makes sense?

---

Because the `USER` directive pretty much has to be hard coded to something, this has to match that, and that something really ought to be your userid for this purpose. 

The good news is that 1000 is very likely to be the uid/gid you get when installing Linux on a desktop (it's mine too) so maybe it's not so bad?